### PR TITLE
my hw jv-spring-security

### PR DESCRIPTION
**Pull request by VolodymyrKachan**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                .getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .collect(Collectors.toList());
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    private String getErrorMessage(ObjectError objectError) {
+        if (objectError instanceof FieldError) {
+            String filed = ((FieldError) objectError).getField();
```
Comment (GolDenik) [inline src/main/java/mate/academy/spring/controller/GlobalExceptionHandler.java:37]:
```suggestion
            String field = ((FieldError) objectError).getField();
```

File: src/main/java/mate/academy/spring/dto/request/UserRequestDto.java [hunk 2]
```diff
@@ -1,8 +1,23 @@
 package mate.academy.spring.dto.request;
 
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import mate.academy.spring.validation.Email;
+import mate.academy.spring.validation.FieldsValueMatch;
+
+@FieldsValueMatch(first = "password", second = "repeatPassword")
 public class UserRequestDto {
+    @NotNull(message = "Email can't is empty")
```
Comment (GolDenik) [inline src/main/java/mate/academy/spring/dto/request/UserRequestDto.java:10]: You don't need this annotation
```suggestion
```

File: src/main/java/mate/academy/spring/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -4,24 +4,24 @@
...
 @Service
 public class UserServiceImpl implements UserService {
-    private static final int SALT_LENGTH = 10;
     private final UserDao userDao;
+    private final PasswordEncoder passwordEncoder;
 
-    public UserServiceImpl(UserDao userDao) {
+    public UserServiceImpl(UserDao userDao, PasswordEncoder passwordEncoder) {
         this.userDao = userDao;
+        this.passwordEncoder = new BCryptPasswordEncoder();
```
Comment (GolDenik) [inline src/main/java/mate/academy/spring/service/impl/UserServiceImpl.java:18]:
```suggestion
        this.passwordEncoder = passwordEncoder;
```

**Review by GolDenik — CHANGES_REQUESTED** [review]

**Review by GolDenik — APPROVED** [review]

File: src/main/java/mate/academy/spring/dto/request/UserRequestDto.java [hunk 4]
```diff
@@ -1,8 +1,22 @@
 package mate.academy.spring.dto.request;
 
+import javax.validation.constraints.Size;
+import mate.academy.spring.validation.Email;
+import mate.academy.spring.validation.FieldsValueMatch;
+
+@FieldsValueMatch(first = "password", second = "repeatPassword")
 public class UserRequestDto {
+
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/dto/request/UserRequestDto.java:9]: remove empty line

**Review by Sofasmile — APPROVED** [review]

**Review by VolodymyrKachan — COMMENTED** [review]
God Job!

**Pull request closed without merging**